libraries/chain/transaction.cpp

```diff
@@ -468,18 +476,22 @@ static prunable_transaction_data make_prunable_transaction_data( bool legacy, co
 packed_transaction::packed_transaction(const signed_transaction& t, bool legacy, compression_type _compression)
  : compression(_compression),
    prunable_data(make_prunable_transaction_data(legacy, t, _compression)),
-   packed_trx(pack_transaction(t, compression)),
+   packed_trx(pack_transaction(t, estimated_size, compression)),
    unpacked_trx(t),
    trx_id(unpacked_trx.id())
-{}
+{
+   estimated_size = estimated_size * 2 + get_prunable_size();
```

Comment:
Shouldn't the `get_prunable_size()` also be doubled since it is also duplicated in `unpacked_trx`?

Reply:
Well, I guess now it's not since `unpacked_trx` is now a `transaction` and not an `signed_transaction`.

Reply:
It still is because at least for `full_legacy` we store it both packed & unpacked.

Reply:
Sorry, I accidentally deleted the comment you responded to (I thought I was deleting a duplicate one elsewhere not the one in this context).

I said I guess it should go back to not being doubled now that `unpacked_trx` is a `transaction`. Then you pointed out that `full_legacy` has additional data. That said, I don't think it should just double `get_prunable_size` because it would be double the signatures unnecessarily as well. I suggested what I recommend to do instead [here](https://github.com/EOSIO/eos/pull/8900#discussion_r404206245).